starter_code/src/movies.js

```diff
@@ -2,14 +2,119 @@
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+function orderByYear(movieList) {
+  let sortedMovies = movieList.slice().sort(function(a, b) {
+    if (a.year === b.year) {
+      return a.title.localeCompare(b.title);
+    }
+    return a.year - b.year;
+  });
+  return sortedMovies;
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(movieList) {
+  let filteredMovies = movieList.filter(function(movie) {
+    if (
+      movie.director === "Steven Spielberg" &&
+      movie.genre.includes("Drama")
+    ) {
+      return true;
+    }
+  });
+  return filteredMovies.length;
+}
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(movieList) {
+  let top20 = movieList
+    .map(function(movie) {
+      return movie.title;
+    })
+    .slice()
+    .sort()
+    .filter(function(movie, index) {
+      if (index < 20) {
+        return true;
+      }
+    });
+  return top20;
+}
+
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movieList) {
+  let ratesTotal = movieList
+    .filter(function(movie) {
+      if (movie.rate) {
+        return true;
+      }
+    })
+    .map(function(movie) {
+      return movie.rate;
+    })
+    .reduce(function(accum, rating) {
+      return accum + rating;
+    }, 0);
+  return Number((ratesTotal / movieList.length).toFixed(2));
+}
+
 // Iteration 5: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(movieList) {
+  let dramaMovieList = movieList.filter(function(movie) {
+    if (movie.genre.includes("Drama")) {
+      return true;
+    }
+  });
+  return ratesAverage(dramaMovieList);
+}
+
 // Iteration 6: Time Format - Turn duration of the movies from hours to minutes
 
+function turnHoursToMinutes(movieList) {
+  let minutesMovieList = movieList.map(function(movie) {
+    if (!movie.duration.includes("h")) {
+      return {
+        ...movie,
+        duration: Number(movie.duration[0] + movie.duration[1])
```

Comment:
I'm pretty sure that movie.duration[0] + movie.duration[1] is basically string concatenation and then the string is turned into a number. You can instead use something called substring() or slice() to make it look a bit better. However, I may be incorrect here